app/models/katello/content_view_puppet_environment.rb

```diff
@@ -34,8 +34,12 @@ def content_type
       Repository::PUPPET_TYPE
     end
 
-    def puppet?
-      true
+    def unprotected
+      false
+    end
+
+    def checksum_type
+      nil
     end
```

Comment:
To clarify, leave unprotected as it is, and see if you can modify the code so it doesn't call checksum_type